app/models/concerns/profile_fields.rb

```diff
@@ -38,6 +38,24 @@ def canonical?
       parent_profile_id.blank?
     end
 
+    def rules_and_rule_groups
+      custom_arrange(conflicting_relationships(rules, rule_groups), required_relationships(rules, rule_groups))
+    end
+
+    def custom_arrange(conflicts, requires)
+      arranged_rule_groups = rule_groups.includes(:rules).arrange_serializable do |parent, children|
+        rule_children = parent.rules.map do |r|
+          { 'rule' => r, 'requires' => requires[r], 'conflicts' => conflicts[r] }
```

Comment:
For requires and conflicts we might only need the `ID`s or `ref_id`s, for now keep it as it is, but we might need to discuss it with FE later. Again, this is just a note, otherwise the format should be fine...

Reply:
Would this help with performance? If you would not have to request the requires/conflicts and just pass the IDs?

Reply:
We talked about this on our call but just want to add a comment about it here: The reason we need to return the entire object is because it can be either a Rule or a RuleGroup so if we only returned the ID for requires/conflicts we wouldn't know the type.